### ISSUE TITLE
First example from "getting started" page: Lack of reproducibility (?)

### ISSUE THREAD

**Issue by JensDotK** [stan-dev/rstan#408]
#### Summary:
I cannot reproduce results from first example (example 1: Eight Schools).
Repreating the simulation several times reveals pretty similar values. 
Am I doing something wrong or is it because of changes in rstan?

#### Description:
Output values of "my" model and tutorial model differ more or less, e.g. values of lp__ are pretty unequal (-5.0 vs -39.5).

Tutorial output refers to NUTS2 as sampler, reproduced code seems to use NUTS(diag_e), but I have no clue whether it matters.

#### Reproducible Steps:
I set up a stan file called _8schools.stan_ in working directory. 
As suggested in tutorial, it contains following stan code.

```
data {
  int<lower=0> J; // number of schools 
  real y[J]; // estimated treatment effects
  real<lower=0> sigma[J]; // s.e. of effect estimates 
}
parameters {
  real mu; 
  real<lower=0> tau;
  real eta[J];
}
transformed parameters {
  real theta[J];
  for (j in 1:J)
    theta[j] = mu + tau * eta[j];
}
model {
  target += normal_lpdf(eta | 0, 1);
  target += normal_lpdf(y | theta, sigma);
}
```

In RStudio, I sucessfully ran following script:
```
schools_dat <- list(J = 8, 
                    y = c(28,  8, -3,  7, -1,  1, 18, 12),
                    sigma = c(15, 10, 16, 11,  9, 11, 10, 18))

fit <- stan(file = '8schools.stan', data = schools_dat, 
            iter = 1000, chains = 4)
```

This step is successful, however, I get a warning message (just for the sake of completeness):

```
Warning messages:
1: In readLines(file, warn = TRUE) :
  incomplete final line found on ~\8schools.stan'
2: There were 2 divergent transitions after warmup. Increasing adapt_delta above 0.8 may help. See
http://mc-stan.org/misc/warnings.html#divergent-transitions-after-warmup 
3: Examine the pairs() plot to diagnose sampling problems
```

Then I get results via
```
print(fit, digits = 1)
```

#### Current Output:
```
> print(fit, digits = 1)
Inference for Stan model: stan-5bc594b34a2.
4 chains, each with iter=1000; warmup=500; thin=1; 
post-warmup draws per chain=500, total post-warmup draws=2000.

          mean se_mean  sd  2.5%   25%   50%   75% 97.5% n_eff Rhat
mu         7.6     0.1 4.9  -2.0   4.5   7.7  10.7  17.3  1348    1
tau        6.5     0.2 5.5   0.3   2.4   5.2   9.1  20.8   736    1
eta[1]     0.4     0.0 0.9  -1.6  -0.2   0.4   1.0   2.2  2000    1
eta[2]     0.0     0.0 0.9  -1.7  -0.6   0.0   0.6   1.8  2000    1
eta[3]    -0.2     0.0 0.9  -1.9  -0.8  -0.2   0.4   1.6  2000    1
eta[4]     0.0     0.0 0.9  -1.7  -0.6   0.0   0.6   1.8  2000    1
eta[5]    -0.3     0.0 0.9  -2.1  -0.9  -0.4   0.2   1.5  2000    1
eta[6]    -0.2     0.0 0.9  -1.9  -0.8  -0.2   0.4   1.5  2000    1
eta[7]     0.4     0.0 0.9  -1.6  -0.2   0.4   1.0   2.2  2000    1
eta[8]     0.1     0.0 0.9  -1.8  -0.6   0.1   0.7   1.9  2000    1
theta[1]  11.1     0.2 8.2  -2.1   5.7  10.1  15.1  32.1  1630    1
theta[2]   7.7     0.1 6.4  -5.3   3.9   7.8  11.5  20.8  2000    1
theta[3]   5.9     0.2 7.4 -10.7   1.9   6.5  10.5  19.9  2000    1
theta[4]   7.7     0.1 6.6  -5.5   3.6   7.8  11.6  20.8  2000    1
theta[5]   5.0     0.1 6.2  -8.3   1.4   5.4   9.2  15.8  2000    1
theta[6]   5.8     0.1 6.5  -8.6   1.9   6.3  10.0  17.2  2000    1
theta[7]  10.5     0.1 6.7  -1.0   6.1   9.9  14.0  25.1  2000    1
theta[8]   8.3     0.2 7.7  -6.8   3.8   7.9  12.5  24.7  2000    1
lp__     -39.5     0.1 2.7 -45.8 -41.1 -39.2 -37.6 -35.0   602    1

Samples were drawn using NUTS(diag_e) at Fri May 05 14:16:23 2017.
For each parameter, n_eff is a crude measure of effective sample size,
and Rhat is the potential scale reduction factor on split chains (at 
convergence, Rhat=1).
```

#### Expected Output:
```
> print(fit, digits = 1)
Inference for Stan model: schools_code.
4 chains, each with iter=1000; warmup=500; thin=1; 
post-warmup draws per chain=500, total post-warmup draws=2000.

         mean se_mean  sd  2.5%  25%  50%  75% 97.5% n_eff Rhat
mu        7.9     0.2 4.9  -2.1  4.5  7.9 11.0  17.8   422    1
tau       6.3     0.3 5.0   0.2  2.5  5.2  8.9  18.7   214    1
eta[1]    0.4     0.0 0.9  -1.5 -0.2  0.4  1.0   2.1   928    1
eta[2]    0.0     0.0 0.9  -1.8 -0.6  0.0  0.5   1.8  1640    1
eta[3]   -0.2     0.0 1.0  -2.1 -0.8 -0.2  0.4   1.8  1243    1
eta[4]    0.0     0.0 0.9  -1.7 -0.6  0.0  0.6   1.7  1421    1
eta[5]   -0.3     0.0 0.9  -2.0 -1.0 -0.4  0.3   1.5   883    1
eta[6]   -0.2     0.0 0.9  -2.0 -0.8 -0.2  0.4   1.6   926    1
eta[7]    0.4     0.0 0.9  -1.4 -0.2  0.4  0.9   2.1   969    1
eta[8]    0.1     0.0 1.0  -1.8 -0.6  0.1  0.7   2.0  1365    1
theta[1] 11.4     0.3 8.1  -1.4  5.9 10.3 15.2  30.6   574    1
theta[2]  7.7     0.2 6.1  -3.7  3.9  7.8 11.4  19.5   762    1
theta[3]  5.8     0.3 7.9 -12.1  1.8  6.5 10.5  19.9   715    1
theta[4]  8.0     0.2 6.5  -5.4  3.9  8.1 12.3  20.2   977    1
theta[5]  5.0     0.3 6.7 -10.3  1.3  5.7  9.5  16.5   667    1
theta[6]  6.0     0.2 6.6  -8.4  2.0  6.2 10.2  18.6   976    1
theta[7] 10.8     0.3 6.8  -1.1  6.2 10.2 14.9  26.0   596    1
theta[8]  8.6     0.3 7.9  -6.1  4.0  8.1 12.6  27.7   629    1
lp__     -5.0     0.1 2.6 -10.7 -6.6 -4.8 -3.1  -0.5   367    1

Samples were drawn using NUTS2 at Fri Apr 12 22:09:54 2013.
For each parameter, n_eff is a crude measure of effective sample size,
and Rhat is the potential scale reduction factor on split chains (at 
convergence, Rhat=1).
```

Comparing both outputs show different results.
In addition, tutorial version uses NUTS2; in "my" model, samples are drawn via NUTS(diag_e).

#### RStan Version:
> packageVersion("rstan")
[1] ‘2.15.1’

#### R Version:
> R.version.string
[1] "R version 3.2.5 (2016-04-14)"

#### Operating System:
Win7, SP 1

**Comment by jgabry** [comment]
I think the output from the example on the "getting started" page is out-of-date (looks like it's from 2013) and maybe there's also a typo. I get the same thing you do when I run it. I will update the output to a 2017 version. 

Oh, and the NUTS2 vs NUTS(diag_e) thing doesn't matter. Well, it's true that the algorithm has changed a bit over the years, but it's nothing you need to worry about here. The "diag_e" is referencing  the diagonal mass matrix ("diag") and the Euclidean metric ("e").

**Comment by jgabry** [comment]
Actually I bet the difference in lp__ is due to normalizing constants not being dropped. The results on the Getting Started page are probably from using the y ~ normal(...) syntax, which drops normalizing constants, whereas the the target += normal_lpdf(y | ...) version doesn't drop the constants. When the model code was changed in the example we probably forgot to update the results. Doesn't affect the parameter inferences (which look the same to me up to RNG noise pretty much). I just updated the output on the Getting Started page. 

Thanks for pointing this out!
